client.go

```diff
@@ -66,6 +71,25 @@ type Client struct {
 	Options []ClientOption
 }
 
+func (c *Client) umask() os.FileMode {
+	if c == nil {
```

Comment:
I find it a bit odd handling the receiver, `c`, being nil.  It may mask bugs where callers are operating on a `nil` `Client` rather than a valid reference and getting inconsistent masks.

Reply:
agreed! `get_file_test` tests are written in such a way that the client is nil, and I'm not sure if consumers of this library are counting on the current behavior of `FileGetter` (or the other substructs). It may just be concealing bugs in the callers...